Area module

FranckZnt19 opened the pull request:
# Description
 - Creation area CRUD web 
 - Creation area API only index for mobile deployment
 - Testing with rspec
 
 # Testing
run the following commands in your terminal:
1. bundle install
2. yarn install
3. rails db:create
4. rails db:migrate
5. rails db:test:prepare
6. bundle exec rspec
  
  ## Screenshots 
![imagen](https://user-images.githubusercontent.com/55332332/114125809-cddcb800-98bc-11eb-8c11-1b10bd6aa103.png)

Commit:
Area web crud module
The area module was created to identify which area a place belongs to within an institution, allowing to create, read, update and delete only within the web.

Commit:
Area API
The area API was created to identify which area a place belongs to within an institution, allowing only the existing data to be read within the system for mobile implementation.

FranckZnt19 committed:
Merge branch 'master' into feature#3-area-crud

ArielMD commented on alerta-covid/spec/requests/areas_spec.rb:
```diff
@@ -0,0 +1,103 @@
+
+RSpec.describe "/areas", type: :request do
+  
+  let!(:institution) { FactoryBot.create(:institution) }
+  let!(:area) { FactoryBot.create(:area, institution_id: institution.id) }
+  let(:valid_attributes) { { institution_id: institution.id, name: "test area" } }
+  let(:invalid_attributes) { { institution_id: nil, name: nil } }
```
Nice job!!!

ramasoftdev replied:
When we're using FactoryBot just place create or build.
`Example let!(:institution) { create(:institution) }`

ArielMD reviewed (dismissed):
Everything is fine, the code improved a lot

EduardoGHdez commented on alerta-covid/db/schema.rb:
```diff
@@ -10,7 +10,7 @@
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2021_04_08_184023) do
+ActiveRecord::Schema.define(version: 2021_03_30_055117) do
```
Just wonder what happened here?

Is this actually changing to the latest schema version?

Did you try to do `git pull` in your branch?

FranckZnt19 replied:
When the issue of some bug fixes was first approved on the master, the PR had a merge conflict on that line only, so when solving it from the github website, it performed automatic master merge.😅  (I have to practice this part more)

EduardoGHdez reviewed (dismissed):
Pls, just make sure we are not messing with the DB schema version, besides this...

Congratulations!

Nice PR, great commits, and super clean contribution 🔥 💯 

This is awesome :shipit: 


Thanks for going the extra-mile 🏅

EduardoGHdez commented on alerta-covid/spec/requests/api/v1/areas_spec.rb:
```diff
@@ -0,0 +1,16 @@
+RSpec.describe "/areas", type: :request do
+  
+  let!(:institution) { FactoryBot.create(:institution) }
+  let!(:area) { FactoryBot.create(:area, institution_id: institution.id) }
```
You do not need to be that explicit when using `FactoryBot`, you can use `create` directly 😄 

Try it 👍🏽

FranckZnt19 replied:
Thank you I will take this observation into account for new Issues. 😅

ramasoftdev commented on alerta-covid/config/application.yml:
```diff
@@ -0,0 +1,17 @@
+#   stripe_api_key: sk_live_EeHnL644i6zo4Iyq4v1KdV9H
+#   stripe_publishable_key: pk_live_9lcthxpSIHbGwmdO941O1XVU
+
+development:
+  username_postgres: frania
+  password_postgres: "12345"
+  port_postgres: "5432"
+  host_postgres: localhost
```
It's recommendable to cut this.

EduardoGHdez replied:
Excellent Call out!

Can you pls remove this. 
It's not necessary and I'd not recommend adding personal configuartions here

ramasoftdev reviewed (dismissed):
All is ok, and suggestions are more for add and handle in future issues. Nice job

Commit:
Suggested Modifications
Changes were made in the creation of factory for the random and fictitious data of the tests, personal credentials were also removed.

ramasoftdev reviewed (approved):
LGTM

mariormt17 reviewed (approved):
Excellent job my friend, with those comments the code was very good😁
